src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,42 @@
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] container;
+
+    public MateStack() {
+        container = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        if (size == container.length) {
+            resize();
+        }
+        container[++size] = value;
```

Comment:
On which position in the array you will add the first element?

Reply:
Oops. At first position. Corrected